core/src/impl/Kokkos_Half_FloatingPointWrapper.hpp

```diff
@@ -839,27 +839,139 @@ class alignas(FloatType) floating_point_wrapper {
     return tmp_lhs < tmp_rhs;
   }
 
+  template <class T>
+  KOKKOS_FUNCTION friend std::enable_if_t<std::is_convertible_v<T, float> &&
+                                              (std::is_same_v<T, float> ||
+                                               std::is_same_v<T, double>),
+                                          bool>
+  operator<(floating_point_wrapper lhs, T rhs) {
+#ifdef KOKKOS_HALF_IS_FULL_TYPE_ON_ARCH
+    // CAUTION: this may result in different implicit casting across backends
+    return lhs.val < rhs;
+#else
```

Comment:
Looks like for `Cuda` there is only the `operator<(__half, __half)` overload.

Reply:
Yes